src/main/java/Storage.java

```diff
@@ -0,0 +1,201 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A Storage class that deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+
+    private File file;
+
+    /**
+     * Creates a Storage object.
+     *
+     * @param filePath A string that represents the directory of a file.
+     */
+    public Storage(String filePath) {
+        file = new File(filePath);
+    }
+
+    /**
+     * Updates the task in the file to be marked as done.
+     *
+     * @param number An int representing the task number.
+     */
+    public void updateDone(int number) {
+        try {
+            String s = "";
+            Scanner scanFile = new Scanner(file);
+            for (int i = 1; i <= TaskList.size(); i++) {
+                if (i == number) {
+                    if (i == TaskList.size()) {
+                        s = s + scanFile.nextLine().replaceFirst("0", "1");
+                    } else {
+                        s = s + scanFile.nextLine().replaceFirst("0", "1") + System.lineSeparator();
+                    }
+                } else {
+                    if (i == TaskList.size()) {
+                        s = s + scanFile.nextLine();
+                    } else {
+                        s = s + scanFile.nextLine() + System.lineSeparator();
+                    }
+                }
+            }
+            FileWriter fw = new FileWriter("data/duke.txt");
+            fw.write(s);
+            fw.close();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number.");
+        } catch (FileNotFoundException e) {
+            System.out.println("File does not exist");
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Loads the tasks in the file into an ArrayList of tasks
+     *
+     * @return An ArrayList containing tasks from the file.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                String type = sc.next();
+                switch (type) {
```

Comment:
- Another good place to consider the use of 👍 enums 👍  
- Could also check for validity of storage data (e.g. if the file is corrupted / of the wrong format ) and throw the appropriate exceptions if the conditions are not met.